float64 data
time stamp